judge/views/user.py

```diff
@@ -335,6 +338,59 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class WCIPEGMergeActivate(LoginRequiredMixin, TitleMixin, FormView):
+    template_name = 'user/wcipeg-merge-activate.html'
+    form_class = WCIPEGMergeActivationForm
+
+    def merge(self, from_user, to_user):
+        from_user.merge_user = to_user
+        from_user.save()
+        with transaction.atomic():
+            Submission.objects.filter(user=from_user).update(user=to_user)
+            Comment.objects.filter(user=from_user).update(user=to_user)
+            CommentVote.objects.filter(voter=from_user).update(voter=to_user)
+
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            token = 'wcipeg_%d_%s' % (int(kwargs['pk']), request.user.email)
+            hmac_token = hmac.new(force_bytes(settings.SECRET_KEY), msg=token.encode('utf-8'), digestmod='sha256')
+            if not hmac.compare_digest(hmac_token.hexdigest(), kwargs['token']):
+                raise PermissionDenied()
+        return super(WCIPEGMergeActivate, self).dispatch(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        self.merge(self.request.user, User.objects.get(pk=self.kwargs['pk']))
```

Comment:
Aren't we dealing with a `PegUser`/`Profile` and `Profile` object instead of 2 `User` objects?

e.g.
```py
from_user.merge_user = to_user
```

`from_user` is a `PEGUser` object and `to_user` should be a `Profile`.

Reply:
Oh yep, you're right -- didn't realize the consequences when I made the `PEGUser` model.